src/index.js

```diff
@@ -24,9 +43,80 @@ const products = [
 ]
 // Target the ul element in index.html and render the above list
 
+const ulList = document.querySelector('.product-list')
+
+products.map((prod) => {
+  console.log('prod', prod)
+  const li = document.createElement('li')
+  const productDescription = `${prod.name}: ${prod.category}`
+  console.log('description', productDescription)
+  const textNode = document.createTextNode(productDescription)
+
+  li.appendChild(textNode)
+  ulList.appendChild(li)
+})
+​
+
+
+
 // 7. Target the button in index.html, attach an event listener to it. Every time we click
 // on the button, it will toggle the product list. Show => hide, hide => show
 
+const button = document.querySelector('.toggle')
+button.addEventListener('click', () => {
+  const computedStyle = window.getComputedStyle(ulList)
+  const currentDisplay = computedStyle.getPropertyValue('display')
+
+  //const currentDisplay = ulList.style.display
+  //console.log('current', computedStyle.getPropertyValue('display'))
+  //ulList.style.display = 'block'
+  if (currentDisplay === 'block') {
+    ulList.style.display = 'none'
+  } else {
+    ulList.style.display = 'block'
+  }
+})
+
 // 8. Target the input element and attach an event listener to it.
 // Once we click search button, you need to use the text input to search for the relevant product name from the list above
 // and show it in the <span> tag
+
+const input = document.querySelector('.search-input')
+const form = document.querySelector('.product-form')
+const spanResult = document.querySelector('.result')
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault()
+  console.log('submit')
+
+  const formData = new FormData(form)
+  const formValues = Object.fromEntries(formData)
+  console.log('form values', formValues)
+
+
+  //products = [{}, {}, {}] is an array of objects
+  const foundProduct = products.find(() => {
```

Comment:
Good job. A sorter form of this is 
`const foundProduct = products.find((prod)=> prod.name === formValues.product))`